pulp_smash/tests/rpm/cli/test_copy_units.py

```diff
@@ -181,3 +270,131 @@ def test_copy_langpacks(self):
                 _count_langpacks(self.cfg, self.repo_id),
                 0,
             )
+
+
+class CopyAndPublishTwoVersionsRepoTestCase(CopyBaseTestCase):
+    """Test whether a repo can copy two versions of RPMs and publish itself.
+
+    This test targets `Pulp Smash #311`_. The test steps are as follow:
+
+    1. Create a repo1 and sync from upstream.
+    2. Find a RPM that has two different versions in this repo.
+    3. Create a new repo2 and copy the older of the RPM version into it.
+    4. Publish the repo2 as a host.
+    5. Copy a newer version into the repo2 and re-publish.
+    6. Yum update on the client and verify update would succeed.
+
+    Note that the 1st repo `_REPO_ID` has been synced in `setUpModule()`,
+    while the 2nd repo `repo_id` has been created by base class without
+    a feed.
+
+    .. _Pulp Smash #311: https://github.com/PulpQE/pulp-smash/issues/311
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create two repositories and synchronize the 1st repo only."""
+        super(CopyAndPublishTwoVersionsRepoTestCase, cls).setUpClass()
+        # Retrieve all modules with multiple versions in the repo1.
+        cls.rpm_dict = {
+            key: value for (key, value)
+            in _get_rpm_names_versions(cls.cfg, _REPO_ID).items()
+            if len(value) > 1
+        }
+        assert len(cls.rpm_dict) > 0
+        # Choose the first module with multiple versions.
+        cls.rpm_name = cls.rpm_dict.keys()[0]
```

Comment:
Under Python 3, `keys()` returns a view object, which causes this line of code blows up. Please fix.